tests/foreman/cli/test_host.py

```diff
@@ -2558,3 +2559,65 @@ def test_positive_dump_enc_yaml(default_sat):
     assert f'fqdn: {default_sat.hostname}' in enc_dump
     assert f'ip: {default_sat.ip_addr}' in enc_dump
     assert 'ssh-rsa' in enc_dump
+
+
+# -------------------------- HOST TRACE SUBCOMMAND SCENARIOS -------------------------
+def generate_custom_rhel_repo_string(major, minor):
+    domain = settings.repos.rhel_os_repo_host
+    if major == 8:
+        baseurl = (
+            f'{domain}/rhel-{major}/rel-eng/RHEL-{major}/'
+            f'latest-RHEL-{major}.{minor}/compose/AppStream/x86_64/os/'
+        )
+    elif major == 7:
```

Comment:
Just a thought, can't we add this logic in `ContentHost.enable_repo` to enable this repos? 
It seems we already we have this repos in `conf/repos.yml`, so can leverage settings like `settings.repos.rhel7_os` and `settings.repos.rhel8_os.appstream`

Reply:
I will think about this

Reply:
I put it in api utils, but definitely makes sense to move it to content host, therefore you do not need to pass the arguments. So good idea thanks.

But usage of configuration constants you propose is another approach

approach I propose, you create the path, so you need to fix it everytime it will change your approach is better, because `conf/repos.constants` will be up to date, but still the path needs to be parsed because in this case `minor - 1` version is required, so if the path will change and you parse let say based on `/` you can still find yourself in position to fix this method, but you don't need to fix the text

Reply:
@Gauravtalreja1 
at the end i decided to not use `rhel8_os.appstream` because it's not garuanteed that `conf/repos` is same version as `conf/content_host` , right now i have rhel 8.4 and repo appstream for rhel 8.2

I would rather use the 8.3 here instead of  8.1 or 8.2

Reply:
@tstrych Yes, we can keep the same versions in `conf/repos` and `conf/content_host`, updating in `satelliteqe-jenkins` MR#457, we missed this earlier when updating rhel8 versions, thanks!

Also, I just check we've `settings.repos.rhel8_repo` constant pointing to `sesame/pub/rhel8.repo`, now as sesame is off, not sure where we keep that. 

Sorry, I didn't get your point of using the `minor-1` version for repos, but if it is a requirement for the test then its fine, I wouldn't bother you :)

Reply:
thanks to your comments I reworked it to use method that already exists in contenthost class so thanks :)